backend/vouch/tests/test_api_voucher.py

```diff
@@ -144,13 +144,24 @@ def test_auth_204_delete(self):
         """
         self.client.force_authenticate(user=self.user1)
         response = self.client.delete(
-            f"{self.voucher_base_url}{self.voucher.pk}/", format="json"
+            f"{self.voucher_base_url}{self.user2.username}/", format="json"
         )
 
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
         with self.assertRaises(ObjectDoesNotExist):
             Voucher.objects.get(pk=self.voucher.pk)
 
+    def test_auth_404_delete_invalid_user(self):
+        """
+        An authenticated user cannot delete a voucher given by a non-existing
+        user.
+        """
+        self.client.force_authenticate(user=self.user2)
+        response = self.client.delete(f"{self.voucher_base_url}unknown/", format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        Voucher.objects.get(pk=self.voucher.pk)
+
     def test_auth_404_delete_someone_else_voucher(self):
```

Comment:
@GresilleSiffle this test probably doesn't make sense anymore (and it still uses the `pk`).

Reply:
Good catch @sigmike . I'm going to fix it in ithttps://github.com/tournesol-app/tournesol/pull/1091